### PR TITLE
✨Support v1alpha2

### PR DESCRIPTION
**Pull request by vincepri**
This PR updates the repository to be in line with other Cluster API providers and support v1alpha2 API types.

The commits have been separated to provide a better way to review the large changes.

NB: Before merging this in, we should cut a `release-0.1` branch.

### CODE REVIEW

**Comment by jdef** [comment]
gcp build fails due to outdated golang:
```
I0909 19:35:26.190] Kubernetes requires go1.12.1 or greater.
I0909 19:35:26.190] Please install go1.12.1 or later.
W0909 19:35:26.197] Traceback (most recent call last):
W0909 19:35:26.198]   File "/workspace/./test-infra/jenkins/../scenarios/execute.py", line 50, in <module>
W0909 19:35:26.198]     main(ARGS.env, ARGS.cmd + ARGS.args)
W0909 19:35:26.198]   File "/workspace/./test-infra/jenkins/../scenarios/execute.py", line 41, in main
W0909 19:35:26.198]     check(*cmd)
```

**Comment by jdef** [comment]
suspect is here: https://github.com/vincepri/cluster-api-provider-gcp/blob/v1a2/WORKSPACE#L20

**Comment by vincepri** [comment]
@jdef I opened a PR to update the CI containers https://github.com/kubernetes/test-infra/pull/14247

**Comment by timothysc** [comment]
Can you squash to a smaller set of commits.  
1 - key code changes 
2 - vendor goo?  

Otherwise this large of a PR is difficult to review.

**Comment by vincepri** [comment]
@timothysc I tried to capture all the similar changes in different commits. The most important ones to review are at the bottom https://github.com/kubernetes-sigs/cluster-api-provider-gcp/pull/143/commits, namely the "Add v1alpha2 X"

**Comment by sb1975** [comment]
will there be any update on the documentation so that we can test and use the v1alpha2 using GCP.

**Comment by vincepri** [comment]
@sb1975 Yes, once we have a few more items figured out, we'll update the documentation and the related Cluster API Quick Start

**Comment by timothysc** [comment]
I'd recommend tagging master before this merge.

**Comment by vincepri** [comment]
@timothysc Branched off master to `release-0.1` branch, not sure if we should tag/release though, I'll defer to @justinsb

File: api/v1alpha2/gcpcluster_types.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	// Project is the name of the project to deploy the cluster to.
+	Project string `json:"project"`
+
+	// The GCP Region the cluster lives in.
+	Region string `json:"region"`
+
+	// The default zone to create instances in.
+	// If empty, the first zone available in the Region is used.
+	// +optional
+	DefaultZone *string `json:"defaultZone,omitempty"`
```
Comment (justinsb) [inline api/v1alpha2/gcpcluster_types.go]: We probably inherited this from v1alpha1, but it's surprising to see this on the cluster.

File: api/v1alpha2/gcpmachine_types.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+// GCPMachineSpec defines the desired state of GCPMachine
+type GCPMachineSpec struct {
+	// Zone is references the GCP zone to use for this instance.
+	Zone string `json:"zone"`
+
+	// InstanceType is the type of instance to create. Example: n1.standard-2
+	InstanceType string `json:"instanceType"`
+
+	// Subnet is a reference to the subnetwork to use for this instance. If not specified,
+	// the first subnetwork retrieved from the Cluster Region and Network is picked.
```
Comment (justinsb) [inline api/v1alpha2/gcpmachine_types.go]: Again, we probably inherited this, but making this optional feels risky

**Comment by justinsb** [comment]
A bunch of stuff we'll probably want to address that I think is inherited from v1alpha1, but thank you so much for taking the giant first step here!  I don't think it makes sense to try to nitpick on a PR of this size, so let's get it in and then tackle things as we notice them!

File: api/v1alpha2/gcpcluster_types.go (hunk 1 above)
Comment (vincepri) [reply api/v1alpha2/gcpcluster_types.go]: This should have been removed, will file a PR

File: api/v1alpha2/gcpmachine_types.go (hunk 2 above)
Comment (vincepri) [reply api/v1alpha2/gcpmachine_types.go]: It seems that the default behavior I've seen in GCP when you create an instance is to create the network interface in the specified network (fix incoming for that) and pick a subnet from the zone specified. 

Do we prefer to make both Zone and Subnet required or only one of them?

**Pull request merged**